scrunch/datasets.py

```diff
@@ -680,7 +683,42 @@ def __readonly__(self, *args, **kwargs):
     del __readonly__
 
 
-class Dataset(object):
+class DatasetVariablesMixin(collections.Mapping):
```

Comment:
There isn't much logic that cannot be a try/except for the not found alias. 
https://github.com/Crunch-io/scrunch/blob/147f639bfc3a3cc5d28a08bb0d6d2db60d455c70/scrunch/datasets.py#L756